src/event-graphics.ts

```
@@ -0,0 +1,138 @@
+"use strict";
+function renderEvent(eventID: string, eventTitle: string, eventStart: Date, eventEnd: Date) {
+
+    const calendarColumns: HTMLCollection = document.getElementsByClassName("calendar-grid__calendar-column");
+    const firstColumn: HTMLElement = calendarColumns[0] as HTMLElement;
+    const columnHeight: number = firstColumn.getBoundingClientRect().height;
+    //const columnWidth: number = firstColumn.offsetWidth;
+
+    const eventOverlay: HTMLElement = document.getElementsByClassName("week-view__calendar-event-overlay")[0] as HTMLElement;
+
+    const eventTime: number = eventStart.getTime() - new Date(eventStart.toDateString()).getTime(); //ms from start of day
+    const eventDuration: number = eventEnd.getTime() - eventStart.getTime(); //ms event duration
+
+    for (let i = 0; i < calendarColumns.length; i++) {
+        const column: HTMLElement = calendarColumns[i] as HTMLElement;
+        const columnDateString: string | undefined = column.getAttribute("data-calendar-day")?.toString();
+        if (columnDateString === undefined) throw new AttributeError("Cannot get data-calendar-day attribute");
```

I wonder if these errors are handled somewhere. Usually, you throw and do nothing unless there's nothing that can be changed (like there's no user action that could resolve this or, at least, inform that something went wrong on technical level (that's the approach in BM btw though on a platform level with an error boundary).

---

They're not handled - the data attributes are always expected to exist, as they're applied at element creation, so the errors related to attributes are only there because I thought it would be best to have them there just in case rather than suppress a potential error. Should I remove them in this scenario?